Sources/MapboxNavigation/CarPlayManager.swift

```diff
@@ -639,8 +649,23 @@ extension CarPlayManager {
             previewText = customPreviewText
         }
         
+        previewMapTemplate.backButton = defaultTripPreviewBackButton()
         previewMapTemplate.showTripPreviews([modifiedTrip], textConfiguration: previewText)
-        interfaceController.pushTemplate(previewMapTemplate, animated: true)
+        
+        if currentActivity == .previewing {
```

Comment:
In case already in preview mode, CarPlay has the template depth constraint, we should only keep one template for routes preview.